From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrzej Hajda <andrzej.hajda@intel.com>
Date: Thu, 7 Mar 2024 14:47:58 +0100
Subject: drm/xe/xe3p: disable EU thread hang on exception

Xe3p supports out-of-band exception reporting.
To enable it legacy mode (EU thread hang) should be disabled.

v2: use XE_RTP_END_VERSION_UNDEFINED instead of -1

v3: reworked to eudebug v3 (Maciej)

v4: removed the empty bracket at the end of engine_tunings (Maciej)

Signed-off-by: Maciej Patelczyk <maciej.patelczyk@intel.com>
Signed-off-by: Andrzej Hajda <andrzej.hajda@intel.com>
Reviewed-by: Dominik Grzegorzek <dominik.grzegorzek@intel.com>

(cherry-picked from commit ecbfd65606569ad9bfd354f9a51d8666c4a056e2 drm-tip-eudebug)
Signed-off-by: Bommu Krishnaiah <krishnaiah.bommu@intel.com>

---
 drivers/gpu/drm/xe/regs/xe_gt_regs.h | 1 +
 drivers/gpu/drm/xe/xe_tuning.c       | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/xe/regs/xe_gt_regs.h b/drivers/gpu/drm/xe/regs/xe_gt_regs.h
index dd1738a2ace3..f215fa03bb86 100644
--- a/drivers/gpu/drm/xe/regs/xe_gt_regs.h
+++ b/drivers/gpu/drm/xe/regs/xe_gt_regs.h
@@ -556,6 +556,7 @@
 #define   TILEY_LOCALID				REG_BIT(2)
 
 #define ROW_CHICKEN				XE_REG_MCR(0xe4f0, XE_REG_OPTION_MASKED)
+#define   CHICK_HANG_SEND_EXCEPTION		REG_BIT(15)
 #define   UGM_BACKUP_MODE			REG_BIT(13)
 #define   MDQ_ARBITRATION_MODE			REG_BIT(12)
 #define   STALL_DOP_GATING_DISABLE		REG_BIT(5)
diff --git a/drivers/gpu/drm/xe/xe_tuning.c b/drivers/gpu/drm/xe/xe_tuning.c
index bcec40ca2d35..3c88204415e2 100644
--- a/drivers/gpu/drm/xe/xe_tuning.c
+++ b/drivers/gpu/drm/xe/xe_tuning.c
@@ -130,6 +130,10 @@ static const struct xe_rtp_table_sr engine_tunings = XE_RTP_TABLE_SR(
 		       FUNC(xe_rtp_match_first_render_or_compute)),
 	  XE_RTP_ACTIONS(SET(TDL_TSL_CHICKEN2, TILEY_LOCALID))
 	},
+	{ XE_RTP_NAME("Tuning: disable EU thread hang on exception"),
+	  XE_RTP_RULES(GRAPHICS_VERSION_RANGE(3500, XE_RTP_END_VERSION_UNDEFINED)),
+	  XE_RTP_ACTIONS(CLR(ROW_CHICKEN, CHICK_HANG_SEND_EXCEPTION))
+	},
 );
 
 static const struct xe_rtp_table_sr lrc_tunings = XE_RTP_TABLE_SR(
-- 
2.43.0

